circuit/algorithms/src/elligator2/encode.rs

```diff
@@ -97,7 +97,7 @@ impl<E: Environment> Elligator2<E> {
         // Convert the Montgomery element (u, v) to the twisted Edwards element (x, y).
         let x = &u / v;
         let y = (&u - &one) / (u + &one);
-        let encoding = Group::from_xy_coordinates(x, y);
+        let encoding = Group::from_xy_coordinates_unchecked(x, y); // This is safe.
```

Comment:
We have to additionally perform an on-curve check here.

```suggestion
         // This is safe because we perform the
         // on-curve and subgroup checks right below.
        let encoding = Group::from_xy_coordinates_unchecked(x, y);
```

Reply:
I have switched this to the checked variant.